File: circlewidgetlibrary/src/main/java/com/jungthinkr/aaronl/circlewidgetlibrary/CircleWidgetView.java
package com.jungthinkr.aaronl.circlewidgetlibrary;

import android.content.Context;
import android.support.annotation.AttrRes;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.view.ViewGroup;
import android.widget.FrameLayout;

public class CircleWidgetView extends FrameLayout {

    public CircleWidgetView(@NonNull Context context) {
        this(context, null);
    }

    public CircleWidgetView(@NonNull Context context, @Nullable AttributeSet attrs) {
        this(context, null, 0);
    }

    public CircleWidgetView(@NonNull Context context, @Nullable AttributeSet attrs, @AttrRes int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        initProperties();
    }

    void initProperties() {

    }

    @Override
    public boolean onTouchEvent(MotionEvent event) {
        switch(event.getAction()) {
            case MotionEvent.ACTION_POINTER_UP:
                break;
        }
        return super.onTouchEvent(event);
    }

    @Override
    protected void onLayout(boolean changed, int l, int t, int r, int b) {
        super.onLayout(changed, l, t, r, b);
    }
}

File: circlewidgetlibrary/src/main/java/com/jungthinkr/aaronl/circlewidgetlibrary/CircleWidgetView.kt
package com.jungthinkr.aaronl.circlewidgetlibrary

import android.content.Context
import android.graphics.Color
import android.support.annotation.AttrRes
import android.support.v4.content.ContextCompat
import android.util.AttributeSet
import android.view.MotionEvent
import android.widget.FrameLayout

class CircleWidgetView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null, @AttrRes defStyleAttr: Int = 0
) : FrameLayout(context, attrs, defStyleAttr), CircleWidget.CircleWidgetListener {
    interface CircleWidgetViewListener {
        fun onWidgetClick()
    }

    private var isWidgetFocused: Boolean = false
    var circleWidgetViewListener: CircleWidgetViewListener? = null

    init {
        val circleWidgetSize = context.resources.getDimension(R.dimen.circle_widget_size)
        val circleWidget = CircleWidget(this, context, attrs, 0)

        addView(circleWidget)
        circleWidget.layoutParams = LayoutParams(circleWidgetSize.toInt(), circleWidgetSize.toInt())
        circleWidget.setBackgroundColor(Color.RED)
    }

    override fun onTouchEvent(event: MotionEvent): Boolean {
        when (event.action) {
            MotionEvent.ACTION_MOVE -> {

            }

        }
        return super.onTouchEvent(event)
    }

    override fun onWidgetClick() {
        circleWidgetViewListener?.onWidgetClick()
    }

    override fun onWidgetActionDown() {
        isWidgetFocused = true
    }

    override fun onWidgetActionUp() {
        isWidgetFocused = false
    }
}
